onnxruntime/core/providers/cuda/shared_inc/fpgeneric.h

```diff
@@ -174,6 +174,238 @@ inline cublasStatus_t cublasGemmHelper(cublasHandle_t, cublasOperation_t, cublas
+  cublasLtMatmulDesc_t operation_desc = NULL;
+
+  auto clean_desc_A = gsl::finally([&A_desc]() {
+    if (A_desc) {
+      cublasLtMatrixLayoutDestroy(A_desc);
+    }
+  });
+
+  auto clean_desc_B = gsl::finally([&B_desc]() {
+    if (B_desc) {
+      cublasLtMatrixLayoutDestroy(B_desc);
+    }
+  });
+
+  auto clean_desc_C = gsl::finally([&C_desc]() {
+    if (C_desc) {
+      cublasLtMatrixLayoutDestroy(C_desc);
+    }
+  });
+
+  auto clean_matmul_desc = gsl::finally([&operation_desc]() {
+    if (operation_desc) {
+      cublasLtMatmulDescDestroy(operation_desc);
+    }
+  });
+
+  if (Status::OK() != InitializeCublasLtMatmulDescAndOperationHelper(A_desc, lda,
+                                                               transa,
+                                                               B_desc, ldb,
+                                                               transb,
+                                                               C_desc, ldc,
+                                                               data_type,
+                                                               m, n, k,
+                                                               operation_desc,
+                                                               compute_type,
+                                                               scale_type)) {
+    return CUBLAS_STATUS_ALLOC_FAILED;
+  }
+
+  // TODO (hasesh): Allow CublasLtMatmul tuning for clients by allowing them to pass in the workspace and algo of their choice.
+  // According to the cublasLtMatmul documentation, passing in NULL for the algo means that "an implicit heuristics query with
+  // default search preferences will be performed to determine actual algorithm to use". Source: cublasLtMatmul documentation.
+  return cublasLtMatmul(
+      handle, operation_desc, is_compute_16f ? reinterpret_cast<const void*>(alpha) : reinterpret_cast<const void*>(&f_alpha),
+      A, A_desc, B, B_desc,
+      is_compute_16f ? reinterpret_cast<const void*>(beta) : reinterpret_cast<const void*>(&f_beta),
+      C, C_desc,
+      C, C_desc,
+      /*algo*/ NULL, /*workspace memory*/ NULL, /*workspace size*/ 0, stream);
+}
+#else
+inline cublasStatus_t cublasLtMatmulHelper(cublasLtHandle_t handle,
+                                           cublasOperation_t transa,
+                                           cublasOperation_t transb,
+                                           int m, int n, int k,
+                                           const half* alpha,
+                                           const half* A, int lda,
+                                           const half* B, int ldb,
+                                           const half* beta,
+                                           half* C, int ldc,
+                                           cudaStream_t stream) {
+  return CUBLAS_STATUS_NOT_SUPPORTED;
+}
+#endif
+
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
```

Comment:
Technically we can still use CublasLtMatmul in CUDA 10.x with some ugly ifdefs (as the interface is a bit different), But AFAIK- we dropped support for CUDA 10 and it doesn't add much value to support it.